### ISSUE TITLE
SDK Update broke publishing in AspnetCore-Tooling

### ISSUE THREAD

**Issue by wtgodbe** [dotnet/aspnetcore#18324]
AspNetCore-Tooling just took an SDK update (to version `5.0.100-alpha.1.20063.3`) in order to unblock ingesting the latest changes from dotnet/runtime. Here is that PR: https://github.com/dotnet/aspnetcore-tooling/pull/1486. This update leads to the following error while publishing to our blob feeds:

> D:\a\1\s\.packages\microsoft.dotnet.arcade.sdk\5.0.0-beta.20062.1\tools\SdkTasks\PublishArtifactsInManifest.proj(65,5): error : Method not found: 'NuGet.Versioning.NuGetVersion NuGet.Packaging.Core.PackageIdentity.get_Version()'.

Here is a relevant build: https://dev.azure.com/dnceng/internal/_build/results?buildId=482014

Presumably the failing call is here: https://github.com/dotnet/arcade/blob/be2d4ae34b4fcfe35beb9bdaeb7b7990c1147740/src/Microsoft.DotNet.Build.Tasks.Feed/src/PublishArtifactsInManifest.cs#L994-L996

We should figure out what changed in nuget.client, and make the appropriate reaction in Arcade.

CC @riarenas @JohnTortugo @dotnet/aspnet-build @rrelyea

**Comment by wtgodbe** [comment]
@nkolev92 is this related to https://github.com/NuGet/NuGet.Client/pull/2670?

**Comment by wtgodbe** [comment]
Oh, I misread the date on that PR as being in this January, not last January. Red herring, I guess.

**Comment by jkotalik** [comment]
Was there another PR failing for the same reason in Arcade? If so, we should look into that.

I scanned through the binlog and didn't see anything out of the ordinary.

With regards to https://github.com/dotnet/aspnetcore/issues/18324, can we revert the update to arcade to still flow dependencies from dotnet/runtime, or are there other changes in arcade we need?

**Comment by wtgodbe** [comment]
> Was there another PR failing for the same reason in Arcade? If so, we should look into that.

It doesn't appear so. I suppose if Arcade was failing to publish we wouldn't have gotten the update from it.

> With regards to #18324, can we revert the update to arcade to still flow dependencies from dotnet/runtime, or are there other changes in arcade we need?

I'm going to try setting up an internal branch with the Arcade update reverted, to see if it fails in the same way.

**Comment by wtgodbe** [comment]
Here's the test build: https://dev.azure.com/dnceng/internal/_build/results?buildId=482731

**Comment by wtgodbe** [comment]
The build failed even after reverting the arcade change, so this is related to the SDK update, not the Arcade update.

**Comment by Pilchie** [comment]
@dsplaisted @wli3 ?

**Comment by wtgodbe** [comment]
Toolset & SDK use a `5.5.0-preview.2` version of `NuGet.Build.Tasks`, from 3 weeks ago:
https://github.com/dotnet/toolset/blob/master/eng/Version.Details.xml#L72-L74
https://github.com/dotnet/sdk/blob/master/eng/Version.Details.xml#L22-L24

While CLI uses a `5.3.0-rtm` version from 4 months ago:
https://github.com/dotnet/cli/blob/master/eng/Version.Details.xml#L52-L54

Looking at DARC, Toolset & SDK are subscribed to Nuget.Client on the `.Net Core 5 Dev' channel, while CLI is not (though all 3 repos subscribe to Nuget.Client on the release channels). @wli3 @dsplaisted @nguerrera is there a reason for that? We may just need to update nuget.client in the CLI, then flow that through to core-sdk & update this repo's SDK version.

**Comment by wli3** [comment]
There is no CLI for 5.0/master. It was merged to toolset

**Comment by wli3** [comment]
@kartheekp-ms for nuget issue

**Comment by wli3** [comment]
I think we had this issue before, we need to update Arcade's nuget version. There should be a breaking change

**Comment by wtgodbe** [comment]
Arcade uses `Nuget.Versioning` version 4.4.0, `Nuget.Client` version 5.3.0: https://github.com/dotnet/arcade/blob/be2d4ae34b4fcfe35beb9bdaeb7b7990c1147740/eng/Versions.props#L42-L43. @tmat @markwilkie can we update these?

**Comment by wtgodbe** [comment]
@wli3 do you remember where you saw this issue before?

@tmat @markwilkie any thoughts on if it's safe to update the nuget versions in Arcade?

**Comment by wli3** [comment]
I think it is that https://github.com/dotnet/arcade/issues/1965

But there is not much information

**Comment by wtgodbe** [comment]
Discussion is happening in https://github.com/dotnet/arcade/issues/1965 now

**Comment by wtgodbe** [comment]
Here's an internal tooling build, testing to see if using an Arcade w/ update Nuget fixes the issue: https://dev.azure.com/dnceng/internal/_build/results?buildId=484557

**Comment by wtgodbe** [comment]
That didn't work either, still getting the same error:

> D:\a\1\s\.packages\microsoft.dotnet.arcade.sdk\5.0.0-beta.20065.1\tools\SdkTasks\PublishArtifactsInManifest.proj(65,5): error : Method not found: 'NuGet.Versioning.NuGetVersion NuGet.Packaging.Core.PackageIdentity.get_Version()'.

https://dev.azure.com/dnceng/internal/_build/results?buildId=484557

**Comment by wtgodbe** [comment]
This has been resolved
